avocado/core/settings.py

```diff
@@ -89,7 +89,7 @@ class ConfigOption:
     def __init__(self, namespace, help_msg, key_type=str, default=None,
```

Comment:
Hi @clebergnu I'm not sure if I completely understood the problem you are solving here.

IMO, adding this new logic might make things more complex, but maybe I'm missing something.

Currently we already don't require users pass a list in the command line. Take the `--loaders` example for instance. You just need to set the proper `nargs` during the register and let argparse handle this. When calling it, you can call it multiple times or use the list separated by space (if placed at the end)

```
$ avocado run references --store-logging-stream=custom1 --store-logging-stream=custom2 ...
```

```
$  avocado run references --store-logging-stream=custom1 custom2 ...
```

Besides that, `key_type` can be also used and pass a function to evaluate this (default is str).

I just would like to understand better if we cant avoid this change, since we do have many others 'nargs' arguments. And probably we are missing consistency among them (currently we have lists, strings and custom lists just to handle multiple arguments).

Reply:
Sure, the problem is that the *currently expected* syntax is [different](https://github.com/avocado-framework/avocado/blob/master/docs/source/guides/user/chapters/logging.rst#storing-custom-logs) and it matches the same expected syntax of `--show`.